examples/main.cpp

```diff
@@ -361,26 +333,45 @@ int run(const char *device_name,
   if (use_device) ams_device = AMSResourceType::DEVICE;
   AMSExecPolicy ams_loadBalance = AMSExecPolicy::UBALANCED;
   if (lbalance) ams_loadBalance = AMSExecPolicy::BALANCED;
+#endif
 
-  AMSConfig amsConf = {ams_loadBalance,
-                       precision,
-                       ams_device,
-                       dbType,
-                       (AMSPhysicFn)callBack<TypeValue>,
-                       (char *)surrogate_path,
-                       (char *)uq_path,
-                       (char *)db_path,
-                       threshold,
-                       uq_policy,
-                       k_nearest,
-                       rId,
-                       wS};
-
+  // ---------------------------------------------------------------------
+  // setup EOS models
+  // ---------------------------------------------------------------------
+  std::vector<EOS<TypeValue> *> eoses(num_mats, nullptr);
   for (int mat_idx = 0; mat_idx < num_mats; ++mat_idx) {
-    AMSExecutor wf = AMSCreateExecutor(amsConf);
-    workflow[mat_idx] = wf;
-  }
+    EOS<TypeValue> * base;
+    if (eos_name == std::string("ideal_gas")) {
+      base = new IdealGas<TypeValue>(1.6, 1.4);
+    } else if (eos_name == std::string("constant_host")) {
+      base = new ConstantEOSOnHost<TypeValue>(physics_host_alloc.c_str(), 1.0);
+    } else {
+      std::cerr << "unknown eos `" << eos_name << "'" << std::endl;
+      return 1;
+    }
+#ifdef USE_AMS
+    if (/*use_ams*/true) {
+      eoses[mat_idx] = new AMSEOS<TypeValue>(base,
```

Comment:
Are we releasing this memory somewhere?

Reply:
Yes, `AMSEOS` will `delete` the `base` `EOS` and at the end of `main` we remove everything from `eoses`